src/librustc/ty/maps/on_disk_cache.rs

```diff
@@ -64,14 +63,21 @@ impl<'sess> OnDiskCache<'sess> {
         let mut decoder = opaque::Decoder::new(&data[..], start_pos);
         let header = Header::decode(&mut decoder).unwrap();
 
-        let prev_diagnostics: FxHashMap<_, _> = {
+        let prev_diagnostics = {
             let mut decoder = CacheDecoder {
                 opaque: decoder,
                 codemap: sess.codemap(),
                 prev_filemap_starts: &header.prev_filemap_starts,
             };
-            let body = Body::decode(&mut decoder).unwrap();
-            body.diagnostics.into_iter().collect()
+
+            let prev_diagnostics: FxHashMap<_, _> = {
```

Comment:
Why do we have this new variable here? It doesn't serve an obvious purpose. =)

Reply:
It does in later commits where this block is used to limit the scope of the `CacheDecoder`.